reactor-core/src/main/java/reactor/core/publisher/Hooks.java

```diff
@@ -570,7 +571,12 @@ public static void resetOnNextError() {
 	Hooks() {
 	}
 
-	static <T, P extends Publisher<T>> Publisher<T> addAssemblyInfo(P publisher, AssemblySnapshot stacktrace) {
```

Comment:
wonder if we could / should migrate that to `Traces.java` (but that would mean making it `public`), wdyt @smaldini ?

Reply:
1) this is not for tracing
2) here we keep it internal and make public at runtime via byte code instrumentation, I would keep it this way

Reply:
`Traces` is not for tracing either. It's where we construct these "tracebacks" that augment stack _Traces_ .

Reply:
IMO it is for, because, in `Traces`, we get the stack trace and trace back the call site, but here we never operate any "traces", but use the file location